globus_cli/parsing/shell_completion.py

```diff
@@ -0,0 +1,195 @@
+import os
+import shlex
+import click
+
+from globus_cli.safeio import safeprint
+from globus_cli.parsing.hidden_option import HiddenOption
+from globus_cli.parsing.case_insensitive_choice import CaseInsensitiveChoice
+
+SUPPORTED_SHELLS = ['BASH', 'ZSH']
+
+
+def safe_split_line(inputline):
+    parser = shlex.shlex(inputline, posix=True)
```

Comment:
We agree that if users aren't lazy and quote their subshells (which they should!) that this works correctly, so non-issue.